pyfr/plugins/ascent.py

```diff
@@ -0,0 +1,282 @@
+# -*- coding: utf-8 -*-
+
+from collections import defaultdict
+import re
+
+import numpy as np
+
+from pyfr.libs.ascent import Ascent
+from pyfr.libs.conduit import ConduitNode, ConduitWrappers
+from pyfr.mpiutil import get_comm_rank_root
+from pyfr.nputil import npeval
+from pyfr.plugins.base import BasePlugin
+from pyfr.shapes import BaseShape
+from pyfr.util import file_index, file_path, subclass_where
+from pyfr.writers.vtk import BaseShapeSubDiv
+
+
+class AscentPlugin(BasePlugin):
+    name = 'ascent'
+    systems = ['*']
+    formulations = ['dual', 'std']
+
+    def __init__(self, intg, cfgsect, suffix=None):
+        super().__init__(intg, cfgsect, suffix)
+
+        comm, rank, root = get_comm_rank_root()
+        self.nsteps = self.cfg.getint(cfgsect, 'nsteps')
+
+        # Underlying system
+        system = intg.system
+        self._enames = system.ele_types
+        self.dtype = intg.system.backend.fpdtype
+
+        # Underlying system elements class
+        self.elementscls = system.elementscls
+
+        # Load conduit library
+        self.conduit = ConduitWrappers()
+
+        # Expressions to plot and configs
+        cons = self.cfg.items_as('constants', float)
```

Comment:
We only seem to pass this down to `_config_scene`.  Probably easier if it takes care of fetching it locally.